src/EditorFeatures/CSharp/UseAutoProperty/UseAutoPropertyAnalyzer.cs

```diff
@@ -0,0 +1,353 @@
+            if (propertyDeclaration == null)
+            {
+                return;
+            }
+
+            // Need at least a getter.
+            var getAccessor = propertyDeclaration.AccessorList.Accessors.FirstOrDefault(d => d.Kind() == SyntaxKind.GetAccessorDeclaration);
+            if (getAccessor == null)
+            {
+                return;
+            }
+
+            // A setter is optional though.
+            var setAccessor = propertyDeclaration.AccessorList.Accessors.FirstOrDefault(d => d.Kind() == SyntaxKind.SetAccessorDeclaration);
+
+            var semanticModel = symbolContext.Compilation.GetSemanticModel(syntaxReference.SyntaxTree);
+            IFieldSymbol getterField = null, setterField = null;
+            if (!CheckGetter(semanticModel, containingType, getAccessor, ref getterField))
+            {
+                return;
+            }
+
+            // Property and field have to agree on type.
+            if (!property.Type.Equals(getterField.Type))
+            {
+                return;
+            }
+
+            // Don't want to remove constants.
+            if (getterField.IsConst)
+            {
+                return;
+            }
+
+            if (getterField.DeclaringSyntaxReferences.Length != 1)
+            {
+                return;
+            }
+
+            // Field and property should match in static-ness
+            if (getterField.IsStatic != property.IsStatic)
+            {
+                return;
+            }
+
+            if (setAccessor != null && !CheckSetter(semanticModel, containingType, setAccessor, ref setterField))
+            {
+                return;
+            }
+
+            if (getterField != null && setterField != null)
+            {
+                // If there is a getter and a setter, they both need to agree on which field they are 
+                // writing to.
+                if (getterField != setterField)
+                {
+                    return;
+                }
+            }
+
+            var fieldReference = getterField.DeclaringSyntaxReferences[0];
+            var variableDeclarator = fieldReference.GetSyntax(symbolContext.CancellationToken) as VariableDeclaratorSyntax;
+            if (variableDeclarator == null)
+            {
+                return;
+            }
+
+            var fieldDeclaration = variableDeclarator?.Parent?.Parent as FieldDeclarationSyntax;
+            if (fieldDeclaration == null)
+            {
+                return;
+            }
+
+            // Can't remove the field if it has attributes on it.
+            if (fieldDeclaration.AttributeLists.Count > 0)
+            {
+                return;
+            }
+
+            // Looks like a viable property/field to convert into an auto property.
+            analysisResults.Add(new AnalysisResult(property, getterField, propertyDeclaration, variableDeclarator,
+                property.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat)));
```

Comment:
Do we want this to be fully qualified?  In what scenario is the user going to need a fully qualified property to disambiguate?